node/full_node_integration_test.go

```diff
@@ -465,16 +486,16 @@ func TestFraudProofService(t *testing.T) {
 // Creates a starts the given number of client nodes along with an aggregator node. Uses the given flag to decide whether to have the aggregator produce malicious blocks.
 func createAndStartNodes(clientNodes int, isMalicious bool, t *testing.T) ([]*FullNode, []*mocks.Application) {
 	aggCtx, aggCancel := context.WithCancel(context.Background())
+	defer aggCancel()
 	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 	nodes, apps := createNodes(aggCtx, ctx, clientNodes+1, isMalicious, t)
 	startNodes(nodes, apps, t)
-	aggCancel()
-	time.Sleep(100 * time.Millisecond)
-	for _, n := range nodes {
-		require.NoError(t, n.Stop())
-	}
-	cancel()
-	time.Sleep(100 * time.Millisecond)
+	defer func() {
+		for _, n := range nodes {
+			require.NoError(t, n.Stop())
```

Comment:
does a require failure here stop the for loop?
If so, we should switch to an assert.

Reply:
besides this, lgtm 👍